cypress/e2e/webTables.cy.js

```
@@ -1,7 +1,192 @@
+          cy.wrap($rowGroup)
+            .find('.rt-td')
+            .then(($cells) => {
+              const worker = {
+                index,
+                firstName: $cells.eq(0).text(),
+                lastName: $cells.eq(1).text(),
+                age: parseInt($cells.eq(2).text()),
+                email: $cells.eq(3).text(),
+                salary: parseInt($cells.eq(4).text()),
+                department: $cells.eq(5).text()
+              };
+              allWorkers.push(worker);
+            });
+        }
+      })
+      .then(() => {
+        const randomIndex = Math.floor(Math.random() * allWorkers.length);
+        const randomWorker = allWorkers[randomIndex];
+        cy.contains('.rt-tr-group', randomWorker.firstName)
+          .find('[title="Delete"]')
+          .click();
+        cy.contains(
+          '.rt-td', randomWorker.firstName).should('not.exist');
+      });
+  });
+
+  it('should delete all workers', () => {
+    const allWorkers = [];
+    cy.countWorkers(allWorkers)
+      .then(() => {
+        allWorkers.forEach(() => {
+          cy.get(`[title="Delete"]`)
+            .first()
+            .click();
+        });
+
+        cy.get('.rt-noData')
+          .should('be.visible')
+          .and('contain', 'No rows found');
+      });
+  });
+  it('should find worked in search field and edit it', () => {
+    const worker = {
+      firstName,
+      lastName,
+      email,
+      age,
+      salary,
+      department
+
+    };
+    cy.get('#searchBox')
+      .type('Cierra');
+    cy.get('#edit-record-1')
+      .click();
+    cy.typeInputByPlaceholder(firstName, 'First Name');
+    cy.typeInputByPlaceholder(lastName, 'Last Name');
+    cy.typeInputByPlaceholder(email, 'name@example.com');
+    cy.typeInputByPlaceholder(age, 'Age');
+    cy.typeInputByPlaceholder(salary, 'Salary');
+    cy.typeInputByPlaceholder(department, 'Department');
+    cy.clickSubmitButton();
+    cy.assertNewUser(worker);
```

This test is not self-contained and relies on hardcoded data (`'Cierra'`) and a fixed selector (`#edit-record-1`). This makes the test brittle and likely to fail if the initial data changes. A more robust approach is to:
1. Add a new worker with unique data at the start of this test.
2. Search for that specific worker (e.g., by their unique email).
3. Click the edit button on that worker's row and proceed with the test.